src/core/scope.ml

```diff
@@ -194,29 +194,35 @@ let scope : mode -> sig_state -> env -> p_term -> tbox = fun md ss env t ->
     | (false::_   , []     ) ->
         (* NOTE this could be improved with more general implicits. *)
         fatal loc "More arguments are required to instantiate implicits."
+  (* Create a new metavariable of type [TYPE] for a missing domain. *)
+  and _Meta_Type : env -> tbox = fun env ->
+    let vs = Env.to_tbox env in
+    let a = Env.to_prod_box env _Type in
+    let m = _Meta_full (fresh_meta_box a (Array.length vs)) vs in
+    (* Sanity check: only variables of [env] free in [m] if not in RHS. *)
+    match md with
+    | M_RHS(_) -> m
+    | _        ->
+      assert (Bindlib.is_closed (Bindlib.bind_mvar (Env.vars env) m)); m
```

Comment:
Is this check really useful? If it fails, it means that there is a bug in Bindlib, right?